sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -3189,10 +3189,10 @@ class Dataset[T] private[sql](
 
   private[sql] def collectToPython(): Int = {
     EvaluatePython.registerPicklers()
-    withNewExecutionId {
+    withAction("collectToPython", queryExecution) { plan =>
```

Comment:
These changes can cause the behavior changes. Please submit a PR to document it.